pkg/services/app/add.go

```diff
@@ -113,15 +113,36 @@ func (a *App) Add(params AddParams) error {
 		return errors.Wrap(err, "could not generate deploy key")
 	}
 
+	if err = a.setAppHash(params); err != nil {
```

Comment:
Adding this as state on the service will not work long term. The hash would persist between HTTP requests and there would be race conditions between `go` routines.

Reply:
The intention is not to save the Add progress state here, we would need something else like a SyncMap where the app hash could be used as a key(or something similar). The actual intention is just to be a bit more aligned to POO as it helps with understanding. For instance `generateAppManifests` could be a behavior of the App struct (`(a *App) generateAppYaml(params AddParams)`) as it creates this YAML content for the App itself.